packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,665 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+import 'button.dart';
+import 'colors.dart';
+import 'debug.dart';
+import 'flat_button.dart';
+import 'icon.dart';
+import 'icons.dart';
+import 'ink_well.dart';
+import 'material.dart';
+import 'theme.dart';
+import 'typography.dart';
+
+// TODO(dragostis): Missing functionality:
+//   * mobile horizontal mode with adding/removing steps
+//   * alternative labeling
+//   * stepper feedback in the case of high-latency interactions
+
+/// The state of a [Step] which is used to control the style of the circle and
+/// text.
+///
+/// See also:
+///
+///  * [Step]
+enum StepState {
+  /// A step that displays its index in its circle.
+  indexed,
+  /// A step that displays a pencil icon in its circle.
+  editing,
+  /// A step that displays a tick icon in its circle.
+  complete,
+  /// A step that is disabled and does not to react to taps.
+  disabled,
+  /// A step that is currently having an error. e.g. the use has submitted wrong
+  /// input.
+  error
+}
+
+/// The type of the [Stepper]. This determines how the stepper is presented.
+enum StepperType {
+  /// A vertical layout of the steps with their content in-between the titles.
+  vertical,
+  /// A horizontal layout of the steps with their content below the titles.
+  horizontal
+}
+
+const TextStyle _kStepStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.white
+);
+
+/// A material step used in [Stepper]. The step can have a title and subtitle,
+/// an icon within its circle, some content and a state that governs its
+/// styling.
+///
+/// See also:
+///
+///  * [Stepper]
+///  * <https://material.google.com/components/steppers.html>
+class Step {
+  /// Creates a step for a [Stepper].
+  ///
+  /// The [title], [content], and [state] arguments must not be null.
+  Step({
+    @required this.title,
+    this.subtitle,
+    @required this.content,
+    this.state: StepState.indexed,
+    this.isActive: false
+  }) {
+    assert(this.title != null);
+    assert(this.content != null);
+    assert(this.state != null);
+  }
+
+  /// The title of the step. It changes color based on the [state].
```

Comment:
It's not clear what "changes color" means in this context. You could explain how color changes are handled in general in the Step doc.